src/generic_core/social.ts

```diff
@@ -329,6 +330,10 @@ module Social {
     public handleUserProfile = (profile :freedom_Social.UserProfile)
         : Promise<void> => {
       var userId = profile.userId;
+      if (!Firewall.isValidUserProfile(profile, null)) {
+        this.log("got user profile: " + JSON.stringify(profile));
+        ui.showNotification("Failed user profile firewall check");
```

Comment:
ui.showNotification results in a desktop notification being shown to the user..  If we either get this wrong, or if a lot of invalid messages come in over the social network, it could be really annoying.  Maybe it would be better just to write error trace and throw away the UserProfile/ClientState/Message?